node/network/protocol/src/request_response/v1.rs

```diff
@@ -24,18 +24,54 @@ use super::request::IsRequest;
 use super::Protocol;
 
 /// Request an availability chunk.
-#[derive(Debug, Clone, Encode, Decode)]
+#[derive(Debug, Copy, Clone, Encode, Decode)]
 pub struct AvailabilityFetchingRequest {
-	candidate_hash: CandidateHash,
-	index: ValidatorIndex,
+	/// Hash of candidate we want a chunk for.
+	pub candidate_hash: CandidateHash,
+	/// The index of the chunk to fetch.
+	pub index: ValidatorIndex,
 }
 
 /// Receive a rqeuested erasure chunk.
 #[derive(Debug, Clone, Encode, Decode)]
 pub enum AvailabilityFetchingResponse {
-	/// The requested chunk.
+	/// The requested chunk data.
 	#[codec(index = 0)]
-	Chunk(ErasureChunk),
+	Chunk(ChunkResponse),
+	/// Node was not in possession of the requested chunk.
+	#[codec(index = 1)]
+	NoSuchChunk,
```

Comment:
What was your decision-making to introduce a new variant instead of Option?

Reply:
It was the other way round. I had a one variant enum at first and added the second variant later on. Reason was future proofing a bit, if response type is an enum already, we can add more variants later on, without breaking compatibility.

Reply:
There could be other reasons for not delivering the chunk. Things that come to mind: `NotAValidator`, `NotYourChunk`, ... stuff like that.

Reply:
Makes sense